refbox/src/audienceInfo.js

```diff
@@ -167,38 +169,24 @@ class AudienceInfo extends React.Component {
       currentScore: {},
       highlightScore: null,
       standings: [],
-    } 
-  }
-
-  // Websocket use based on 
-  // https://dev.to/finallynero/using-websockets-in-react-4fkp
-  // instance of websocket connection as a class property
-  ws = new WebSocket('ws://localhost:6789')
-
-  componentDidMount() {
-    this.ws.onopen = () => {
-      // on connecting, do nothing but log it to the console
-      console.log('connected')
     }
+    this.onMessage = this.onMessage.bind(this);
+  }
 
-    this.ws.onmessage = evt => {
-      // listen to data sent from the websocket server
-      const message = JSON.parse(evt.data)
-      console.log('Message: ', message);
-      this.setState({dataFromServer: message})
-      if (this.state.arena in message) {
-        this.updateArenaData(message[this.state.arena]);
-      }
-      if ('metadata' in message && this.state.arena in message.metadata) {
-        this.updateMetaData(message.metadata[this.state.arena])
-      }
+  onMessage(evt) {
+    // listen to data sent from the websocket server
+    const message = JSON.parse(evt);
+    console.log('Message: ', message);
+    this.setState({dataFromServer: message});
+    if (this.state.arena in message) {
+      this.updateArenaData(message[this.state.arena]);
     }
-
-    this.ws.onclose = () => {
-      console.log('disconnected')
-      // automatically try to reconnect on connection loss
+    if ('metadata' in message && this.state.arena in message.metadata) {
+      this.updateMetaData(message.metadata[this.state.arena]);
     }
+  }
 
+  componentDidMount() {
```

Comment:
Deprecated?

Reply:
Yes, I don't know react requires this function to be defined.